packages/host/app/commands/search-cards.ts

```diff
@@ -0,0 +1,93 @@
+import { service } from '@ember/service';
+
+import flatMap from 'lodash/flatMap';
+
+import { Filter, assertQuery } from '@cardstack/runtime-common';
+
+import type * as BaseCommandModule from 'https://cardstack.com/base/command';
+
+import HostBaseCommand from '../lib/host-base-command';
+
+import type CardService from '../services/card-service';
+import type RealmServerService from '../services/realm-server';
+
+export class SearchCardsByTypeAndTitleCommand extends HostBaseCommand<
+  BaseCommandModule.SearchCardsByTypeAndTitleInput,
+  BaseCommandModule.SearchCardsResult
+> {
+  description = 'Search for card instances by type and/or title';
+
+  async getInputType() {
+    let commandModule = await this.loadCommandModule();
+    const { SearchCardsByTypeAndTitleInput } = commandModule;
+    return SearchCardsByTypeAndTitleInput;
+  }
+
+  protected async run(
+    input: BaseCommandModule.SearchCardsByTypeAndTitleInput,
+  ): Promise<BaseCommandModule.SearchCardsResult> {
+    if (!input.title && !input.cardType && !input.type) {
+      throw new Error(
+        'At least one of title, cardType, or type must be provided',
+      );
+    }
+    let filter = {} as any;
+    if (input.title) {
+      filter.contains = { title: input.title };
+    }
+    if (input.cardType) {
+      filter.eq = { _cardType: input.cardType };
```

Comment:
because we are using the card type's `displayName` here (that is really what the `_cardType` field is searching against), there is the chance that different CardDef's may coincidentally have the same `displayName`. so this is not necessarily a super precise search. If we wanted to be more precise we should use the `on` field and provide the card's code ref that we are searching against.

Reply:
Wait, what is the difference between `on` and `type` (see line 41-34 below). That is where I am providing a CodeRef.

Reply:
ah! i missed this. `type` is an alias for `on`. generally we use `on` because you can specify it at any level of the filter. but at the top level we provide a `type` reads nice.